src/web/components/StickyBottomBanner/PuzzlesBanner.tsx

```diff
@@ -0,0 +1,238 @@
+import * as emotionTheming from 'emotion-theming';
+import { useHasBeenSeen } from '@root/src/web/lib/useHasBeenSeen';
+import {
+	getWeeklyArticleHistory,
+	logView,
+} from '@root/node_modules/@guardian/automat-client';
+import {
+	shouldHideSupportMessaging,
+	getArticleCountConsent,
+} from '@root/src/web/lib/contributions';
+import {
+	sendOphanComponentEvent,
+	submitComponentEvent,
+} from '@root/src/web/browser/ophan/ophan';
+import { getCookie } from '@root/src/web/browser/cookie';
+import { getZIndex } from '@root/src/web/lib/getZIndex';
+import { trackNonClickInteraction } from '@root/src/web/browser/ga/ga';
+import { WeeklyArticleHistory } from '@root/node_modules/@guardian/automat-client/dist/types';
+import { CanShowResult } from '@root/src/web/lib/messagePicker';
+
+const checkForErrors = (response: Response) => {
+	if (!response.ok) {
+		throw Error(
+			response.statusText ||
+				`SlotBanner | An api call returned HTTP status ${response.status}`,
+		);
+	}
+	return response;
+};
+
+type HasBeenSeen = [boolean, (el: HTMLDivElement) => void];
+
+type BaseProps = {
+	isSignedIn: boolean;
+	contentType: string;
+	sectionName?: string;
+	shouldHideReaderRevenue: boolean;
+	isMinuteArticle: boolean;
+	isPaidContent: boolean;
+	isSensitive: boolean;
+	tags: TagType[];
+	contributionsServiceUrl: string;
+	alreadyVisitedCount: number;
+	engagementBannerLastClosedAt?: string;
+	subscriptionBannerLastClosedAt?: string;
+	weeklyArticleHistory?: WeeklyArticleHistory;
+};
+
+type BuildPayloadProps = BaseProps & {
+	countryCode: string;
+	hasConsentedToArticleCounts: boolean;
+};
+
+type CanShowProps = BaseProps & {
+	asyncCountryCode: Promise<string>;
+	remotePuzzlesBannerConfig: boolean;
+};
+
+// TODO specify return type (need to update client to provide this first)
+const buildPayload = (props: BuildPayloadProps) => {
+	return {
+		tracking: {
+			ophanPageId: window.guardian.config.ophan.pageViewId,
+			platformId: 'GUARDIAN_WEB',
+			clientName: 'dcr',
+			referrerUrl: window.location.origin + window.location.pathname,
+		},
+		targeting: {
+			alreadyVisitedCount: props.alreadyVisitedCount,
+			shouldHideReaderRevenue: props.shouldHideReaderRevenue,
+			isPaidContent: props.isPaidContent,
+			showSupportMessaging: !shouldHideSupportMessaging(props.isSignedIn),
+			engagementBannerLastClosedAt: props.engagementBannerLastClosedAt,
+			subscriptionBannerLastClosedAt:
+				props.subscriptionBannerLastClosedAt,
+			mvtId: Number(getCookie('GU_mvt_id')),
+			countryCode: props.countryCode,
+			weeklyArticleHistory: getWeeklyArticleHistory(),
+			hasOptedOutOfArticleCount: !props.hasConsentedToArticleCounts,
+		},
+	};
+};
+
+const getPuzzlesBanner = (
+	meta: { [key: string]: any },
+	url: string,
+): Promise<Response> => {
+	const json = JSON.stringify(meta);
+	return fetch(url, {
+		method: 'post',
+		headers: { 'Content-Type': 'application/json' },
+		body: json,
+	});
+};
+
+export const canShow = async ({
```

Comment:
Would it be possible to export this function from it's own file? I think the `PuzzleBanner` component should only be that and then if you have another file calls `lib/canShow.ts` then that is it's own stand alone (perhaps more easilly tested?) thing